components/brave_wallet/browser/eth_tx_state_manager.cc

```diff
@@ -243,37 +243,23 @@ EthTxStateManager::GetTransactionsByStatus(TransactionStatus status,
 }
 
 void EthTxStateManager::ChainChangedEvent(const std::string& chain_id) {
-  rpc_controller_->GetNetwork(base::BindOnce(&EthTxStateManager::OnGetNetwork,
+  rpc_controller_->GetChainId(base::BindOnce(&EthTxStateManager::OnGetChainId,
                                              weak_factory_.GetWeakPtr()));
 }
 
 std::string EthTxStateManager::GetNetworkId() const {
+  auto networks = GetAllNetworks(prefs_);
   std::string id;
-
-  switch (network_) {
-    case brave_wallet::mojom::Network::Mainnet:
-      id = "mainnet";
-      break;
-    case brave_wallet::mojom::Network::Rinkeby:
-      id = "rinkeby";
-      break;
-    case brave_wallet::mojom::Network::Ropsten:
-      id = "ropsten";
-      break;
-    case brave_wallet::mojom::Network::Goerli:
-      id = "goerli";
-      break;
-    case brave_wallet::mojom::Network::Kovan:
-      id = "kovan";
-      break;
-    case brave_wallet::mojom::Network::Localhost:
-      id = network_url_;
-      break;
-    case brave_wallet::mojom::Network::Custom:
-    default:
-      NOTREACHED();
+  for (const auto& network : networks) {
+    if (network.chain_id != chain_id_)
+      continue;
+    if (network.rpc_urls.size()) {
+      id = GURL(network.rpc_urls.front()).host();
```

Comment:
Im not sure if it is ok for this @darkdh, let me know how can I test it properly?

Reply:
We don't need url for known networks like mainnet, rinkeby... because our url endpoint might changed then saved transactions won't be found. We only need that for custom url or localhost This is tested in `EthTxStateManagerUnitTest.SwitchNetwork` unittest, by checking the pref we wrote

Reply:
For those know networks, uses `chainName` or `chainId` to index transactions and if localhost port won't change we can also use those instead of url.

Reply:
done @darkdh pls double check